backend/src/integration-test/kotlin/com/tngtech/apicenter/backend/connector/rest/controller/SpecificationControllerIntegrationTest.kt

```diff
@@ -25,19 +23,29 @@ class SpecificationControllerIntegrationTest {
     @Autowired
     private lateinit var mockMvc: MockMvc
 
+    private val userRequestPostProcessor = authentication(JwtAuthenticationToken("user", "irrelevant"))
+    private val otherRequestPostProcessor = authentication(JwtAuthenticationToken("other", "irrelevant"))
+
     @Test
     fun findOneSpecification_shouldReturnSpecification() {
         mockMvc.perform(get("/api/v1/service/b6b06513-d259-4faf-b34b-a216b3daad6a/version/1.0.0")
-            .with(user("user"))
+            .with(userRequestPostProcessor)
             .with(csrf()))
             .andExpect(jsonPath("$.metadata.version", equalTo("1.0.0")))
     }
 
+    @Test
+    fun findOneSpecification_requiresAuthentication() {
+        mockMvc.perform(get("/api/v1/service/b6b06513-d259-4faf-b34b-a216b3daad6a/version/1.0.0")
+                .with(csrf()))
+                .andExpect(status().isNotFound)
```

Comment:
We did, I think quite a bit :) What I remember is a policy of not leaking information about what services exist, if a user doesn't have view permission. Failed views should get 404 instead of anything indicating the operation failure comes from insufficient permission. Am I mistaken?

Reply:
Yes, with that explanation, I still agree :)
I was just wondering if this was the occassion for the discussion, or something different ;)